examples/mask.py

```diff
@@ -132,70 +128,65 @@ def main(*args):
         raise ValueError("Require at least one image file name: non given")
-    print("Press any key to quit")
     pg.init()
-    screen = pg.display.set_mode((640, 480))
-    images = []
-    masks = []
-    for impath in args:
-        images.append(pg.image.load(impath).convert_alpha())
-        masks.append(maskFromSurface(images[-1]))
-
-    numtimes = 10
-    import time
 
-    t1 = time.time()
-    for x in range(numtimes):
-        unused_mask = maskFromSurface(images[-1])
-    t2 = time.time()
+    screen_size = (640, 480)
+    screen = pg.display.set_mode(screen_size)
+    clock = pg.time.Clock()
 
-    print(f"python maskFromSurface :{t2 - t1}")
-
-    t1 = time.time()
-    for x in range(numtimes):
-        unused_mask = pg.mask.from_surface(images[-1])
-    t2 = time.time()
-
-    print(f"C pg.mask.from_surface :{t2 - t1}")
+    images = []
+    masks = []
+    for image_path in args:
+        images.append(pg.image.load(image_path).convert_alpha())
+        masks.append(pg.mask.from_surface(images[-1]))
 
     sprites = []
     for i in range(20):
         j = i % len(images)
-        s = Sprite(images[j], masks[j])
-        s.setPos(
-            (
-                random.uniform(0, screen.get_width()),
-                random.uniform(0, screen.get_height()),
-            )
+        sprite = Sprite(
+            pos=(
+                random.uniform(0, screen_size[0]),
+                random.uniform(0, screen_size[1]),
+            ),
+            vel=(
+                random.uniform(-5, 5),
+                random.uniform(-5, 5),
+            ),
+            surface=images[j],
+            mask=masks[j],
         )
-        s.setVelocity((random.uniform(-5, 5), random.uniform(-5, 5)))
-        sprites.append(s)
-    pg.time.set_timer(pg.USEREVENT, 33)
+        sprites.append(sprite)
+
     while True:
-        event = pg.event.wait()
-        if event.type == pg.QUIT:
-            return
-        elif event.type == pg.USEREVENT:
-
-            # Do both mechanics and screen update
-            screen.fill((240, 220, 100))
-            for i, sprite in enumerate(sprites):
-                for j in range(i + 1, len(sprites)):
-                    sprite.collide(sprites[j])
-            for s in sprites:
-                s.update(1)
-                if s.pos[0] < -s.surface.get_width() - 3:
-                    s.pos[0] = screen.get_width()
-                elif s.pos[0] > screen.get_width() + 3:
-                    s.pos[0] = -s.surface.get_width()
-                if s.pos[1] < -s.surface.get_height() - 3:
-                    s.pos[1] = screen.get_height()
-                elif s.pos[1] > screen.get_height() + 3:
-                    s.pos[1] = -s.surface.get_height()
-                screen.blit(s.surface, s.pos)
-            pg.display.update()
-        elif event.type == pg.KEYDOWN:
-            return
+        for event in pg.event.get():
+            if event.type in (pg.QUIT, pg.KEYDOWN):
+                return
```

Comment:
Other pygame examples use `break` instead of `return`, this should be changed here to stay consistent.

Reply:
That would just break out of the event loop, and using `pygame.event.wait` in this case isn't possible because that would cause the sprites to stop moving